ttyclock.c

```diff
@@ -166,15 +166,10 @@ signal_handler(int signal)
 void
 cleanup(void)
 {
-    if (ttyclock->ttyscr)
-        delscreen(ttyclock->ttyscr);
-
-    if (ttyclock && ttyclock->tty)
-        free(ttyclock->tty);
-    if (ttyclock && ttyclock->option.format)
-        free(ttyclock->option.format);
-    if (ttyclock)
-        free(ttyclock);
+    if (ttyclock.ttyscr)
+        delscreen(ttyclock.ttyscr);
+
+    free(ttyclock.tty);
```

Comment:
we dropped the tty check here... shouldn't this be:

```suggestion
    if (ttyclock.tty)
        free(ttyclock.tty);
```

Reply:
The `free` function does not need any check.